R/hazard_link.R

```diff
@@ -0,0 +1,79 @@
+
+
+#' Hazard Link Class
+#'
+#' Creates a Hazard Link object which combines a longditudinal model to
+#' an OS model
+#'
+#' @slot stan A `StanModule` object as created by [StanModule()] that specifies any Stan code 
+#' required for the link / contribution
+#' @slot contribution Stan code for what this link contriutes to the log hazard in the OS model
+#' @slot parameter The name of link parameter
+#' @examples
+#' HazardLink(
+#'     parameter = "beta_ttg",
+#'     contribution = "beta_ttg * ttg(phi)",
+#'     stan = StanModule(
+#'         functions = "real ttg(real phi) {phi^2 };"
+#'     )
+#' )
+#' @export
+HazardLink <- setClass(
+    "HazardLink",
+    representation = list(
+        "stan" = "StanModule",
+        "contribution" = "character",
+        "parameters" = "character"
+    )
+)
+
+
+#' @importFrom assertthat assert_that
+#' @rdname StanModule-class
+#' @export
+setMethod(
+    f = "initialize",
+    signature = "HazardLink",
+    definition = function(.Object, ..., stan = StanModule(), contribution, parameters) {
```

Comment:
`stan = StanModule()` should we further specify this to produce a default `HazardLink` ?

Reply:
What would the default be?

My thinking is that this is more of a fundamental object line `StanModule()` itself but doesn't really have a natural default outside of its model specific implementations i.e.  `ExponentialHazardLink()`